File: datamodel/src/main/java/datamodel/clinical/Intolerance.java
package datamodel.clinical;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class Intolerance {

    @NotNull
    public abstract String name();

    @NotNull
    public abstract Set<String> doids();

    @NotNull
    public abstract String category();

    @NotNull
    public abstract Set<String> subcategories();

    @NotNull
    public abstract String type();

    @NotNull
    public abstract String clinicalStatus();

    @NotNull
    public abstract String verificationStatus();

    @NotNull
    public abstract String criticality();

}

File: algo/src/main/kotlin/com/hartwig/actin/clinical/datamodel/Intolerance.kt
package com.hartwig.actin.clinical.datamodel

data class Intolerance(
    val name: String,
    val doids: Set<String>,
    val category: String,
    val subcategories: Set<String>,
    val type: String,
    val clinicalStatus: String,
    val verificationStatus: String,
    val criticality: String
)
